### PR TITLE
moved methods to lib/resource and added a version check

### PR DESCRIPTION
**Pull request by pruan-rht**
addresses https://github.com/openshift/verification-tests/pull/127#discussion_r268840474.  @akostadinov  PTAL.

### CODE REVIEW

File: lib/openshift/cluster_operator.rb [hunk 1]
```diff
@@ -6,17 +6,10 @@ def versions(user: nil, cached: true, quiet: false)
...
-      rr.dig('status', 'conditions')
-    end
-
-    # returns a hash simliar to
-    # {"lastTransitionTime"=>2019-03-23 07:33:34 UTC, "status"=>"False", "type"=>"Progressing"}
-    def condition(type: type, user: nil, cached: true, quiet: false)
-      res = self.conditions.select { |c| c['type'] == type }
-      res.first
+    # given a target_version return true if has matching verison else false
+    def has_version?(version: nil, user: nil, cached: true, quiet: false)
```
Comment (akostadinov) [inline lib/openshift/cluster_operator.rb]: `is_version?` sounds better to me, wdyt? Has versions like owning the version which a little bit makes me confused.

File: lib/openshift/resource.rb [hunk 2]
```diff
@@ -279,6 +279,18 @@ def status?(user: nil, status:, quiet: false, cached: false)
...
+      rr.dig('status', 'conditions')
+    end
+
+    # returns a hash simliar to
+    # {"lastTransitionTime"=>2019-03-23 07:33:34 UTC, "status"=>"False", "type"=>"Progressing"}
+    def condition(type: nil, user: nil, cached: true, quiet: false)
+      res = self.conditions.select { |c| c['type'] == type }
+      res.first
+    end
+
```
Comment (akostadinov) [inline lib/openshift/resource.rb]: good

File: lib/openshift/cluster_operator.rb (hunk 1 above)
Comment (pruan-rht) [reply lib/openshift/cluster_operator.rb]: @akostadinov, the use case from @anuragthehatter is that he wants to check if a version exists.. perhaps `exists`?

Comment (pruan-rht) [reply lib/openshift/cluster_operator.rb]: @akostadinov  ok, I changed it to `version_exists?`

**Pull request merged**